digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/CalibrationPointDAO.scala

```diff
@@ -51,8 +51,20 @@ object CalibrationPointDAO {
          PROJECT_LINK_ID = $projectLinkId And PROJECT_ID = $projectId And ABS(LINK_M - $segmentMValue) < $epsilon
        """
 
-    Q.queryNA[(Long, Long, Long, Double, Long)](baseQuery).list.map{
-      case (id,linkId, prId, linkM, addressM) => UserDefinedCalibrationPoint(id,linkId, prId, linkM, addressM)
+    Q.queryNA[(Long, Long, Long, Double, Long)](baseQuery).list.map {
+      case (id, linkId, prId, linkM, addressM) => UserDefinedCalibrationPoint(id, linkId, prId, linkM, addressM)
+    }
+  }
+
+  def findEndCalibrationPoint(projectLinkId: Long, projectId: Long): Seq[UserDefinedCalibrationPoint] = {
+    val baseQuery =
+      s"""
+         Select ID, PROJECT_LINK_ID, PROJECT_ID, LINK_M, ADDRESS_M From CALIBRATION_POINT Where
+         PROJECT_LINK_ID = $projectLinkId And PROJECT_ID = $projectId And ADDRESS_M > 0
```

Comment:
Offering a sugestion, if you want to get the calibration point at the END of a road wouldn't it make MORE sense to do something like:

Select ID, PROJECT_LINK_ID, PROJECT_ID, LINK_M, ADDRESS_M From CALIBRATION_POINT Where PROJECT_LINK_ID = $projectLinkId And PROJECT_ID = $projectId And ADDRESS_M IN (_Select Max(Address_M) rom CALIBRATION_POINT Where PROJECT_LINK_ID = $projectLinkId And PROJECT_ID = $projectId)_